ua_parser/user_agent_parser.py

```diff
@@ -205,12 +209,20 @@ def Parse(user_agent_string, **jsParseBits):
       A dictionary containing all parsed bits
     """
     jsParseBits = jsParseBits or {}
-    return {
+    key = (user_agent_string, repr(jsParseBits))
```

Comment:
I'm not happy with using `repr()` like this, but we need a hashable type to use as a key.